src/playground/services/invoices/retrieve/endpoint.clj

```diff
@@ -16,12 +19,16 @@
 
 (spec/def ::api (spec/keys :req-un [::models.user/id]))
 
-(defn perform [{{:keys [id]} :path-params :keys [db] :as request}]
-  (let [db     (->> db :pool (hash-map :datasource))
-        record (->> (logic/to-query id)
-                    (h/format)
-                    (jdbc/query db)
-                    (first))]
-    (if record
-      (ring-resp/response (view/update-invoice record))
-      {:status 404 :body "Entry not in DB"})))
+(defn perform [{{:keys [id]} :path-params :keys [session db] :as request}]
+  (if (authenticated? session)
```

Comment:
More about code "thinness" : I am not sure how to solve the seperation of concerns concerns, 
The only way I think of will move `buddy.auth/authenticated?` to the logic only to return back here as a different name from the `logic` namespace but the `(if...` will still happen here. What am I missing?

Reply:
Probably the suggested authorization interceptor would address this concern